tests/dummy/app/controllers/application.js

```diff
@@ -177,5 +250,51 @@ export default Ember.Controller.extend({
         const dimensions = this.get('queueDimensions');
         const groupNames = ['interactions'];
         this.set('queueGroups', groupNames.map(name => dimensions.group().reduceCount(item => item[name])));
+    },
+
+    _createStatusDimension() {
+        let content = Ember.get(this, 'statusContent');
+
+        if (this._StatusCrossfilter) {
+            this._StatusCrossfilter.remove();
+            this._StatusCrossfilter.add(content);
+        } else {
+            this._StatusCrossfilter = crossfilter(content);
+        }
+
+        this.set('statusDimension', this._StatusCrossfilter.dimension(d => d[this._format.title]));
+    },
+
+    _createStatusGroup() {
+        const dimensions = this.get('statusDimension');
+
+        // generic color mapping code: this is still here for potential future use
+
+        // const content = this.get('statusContent');
+        // let colorsMap = {}, colorsArray = [], j = 0;
+        // for (let i = 0; i < content.length; i++) {
+        //     let color = content[i][this._format.color];
+        //     if (colorsArray.indexOf(color) === -1) {
+        //         colorsArray.push(color);
+        //         colorsMap[color] = j;
+        //         j++;
+        //     }
+        // }
```

Comment:
Leaving commented out code blocks lying around isn't typically a good practice. Maybe we could just create a JIRA to add this feature and attach the code block to that ticket?

Reply:
Yep, we could do that, but it's not really a feature, just a different way of creating the color map. It'll probably be pretty useful once we actually put this chart in the app, but you're right that this isn't great practice. Maybe I can just put this in the docs as an example of how the color map should work?

Reply:
This code is actually used in heatmap, so I'll leave it out here.